libminifi/src/core/ContentRepository.cpp

```diff
@@ -60,16 +60,29 @@ void ContentRepository::incrementStreamCount(const minifi::ResourceClaim &stream
   }
 }
 
+void ContentRepository::removeFromPurgeList() {
+  for (auto it = purge_list_.begin(); it != purge_list_.end();) {
+    if (removeKey(*it)) {
+      purge_list_.erase(it++);
+    } else {
+      ++it;
+    }
+  }
+}
+
 ContentRepository::StreamState ContentRepository::decrementStreamCount(const minifi::ResourceClaim &streamId) {
   std::lock_guard<std::mutex> lock(count_map_mutex_);
+  removeFromPurgeList();
```

Comment:
the cleanup logic is quite similar to `FlowFileRepository`'s background `flush` thread (and the planned `DatabaseContentRepository`'s background batch cleanup), I think we should move this cleanup task out of `decrementStreamCount`'s responsibilities

Reply:
You are right it shouldn't be the responsibility of `decrementStreamCount`, I moved it to be part of the `remove` in bde565c405e3e9419020445a48d41da561b58729 I'm not sure if there should be a separate thread for this, what do you think?

Reply:
I'm not sure either, I think it is good as it is now
(later, if needed, we could move it to its own thread later or even integrate it into the planned async purge, we'll see)